### PR TITLE
EARTH-000: Finish refactoring for dependencies and twig functions

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW

# Summary
- Finished up refactoring patterns. 
- Did not do variants tho.

# Needed By (Date)
- Sometime before the patterns get worked on some more.

# Urgency
- meh

# Steps to Test

1.  Merge
2. Clear caches
3. Visit patterns page and look for badness

# Affected Projects or Products
- EARTH

# Associated Issues and/or People
- None

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: patterns/organisms/section-tall-filmstrip/section-tall-filmstrip.html.twig [hunk 1]
```diff
@@ -25,7 +25,16 @@
...
+<<<<<<< HEAD
+<<<<<<< HEAD
     {% if not variants.is_expanded and field_count(items) > 4 %}
+=======
+    {{ dump(items) }}
+    {% if not variants.is_expanded and field_count(items) > 4%}
+>>>>>>> 587c99d... fixup! wip.
+=======
+    {% if not variants.is_expanded and field_count(items) > 4 %}
+>>>>>>> 3560aab... EARTH-302: Change the field cound logic so patterns that have theme set to none get a valid count.
```
Comment (pookmish) [inline patterns/organisms/section-tall-filmstrip/section-tall-filmstrip.html.twig]: Conflicts

Comment (pookmish) [inline patterns/organisms/section-tall-filmstrip/section-tall-filmstrip.html.twig]: And a dump()

**Comment by pookmish** [comment]
Should we add some `{{ spaceless }}` while we're in there?

File: patterns/atoms/photo-thumb/photo-thumb.html.twig [hunk 2]
```diff
@@ -4,12 +4,13 @@
  * Photo thumb pattern.
  */
 #}
-<div class="photo-thumb tile-item">
+{% set attributes = attributes.addClass("photo-thumb tile-item") %}
+<div{{ attributes }}>
   {% if link is not empty %}
```
Comment (pookmish) [inline patterns/atoms/photo-thumb/photo-thumb.html.twig]: Shouldn't need the `if` anymore with the `|openlink` filter

File: patterns/molecules/calendar-blocks/calendar-blocks.html.twig [hunk 3]
```diff
@@ -4,19 +4,20 @@
...
       'day': item.day,
       'month': item.month,
-      'subtitle': item.subtitle
+      'subtitle': item.subtitle,
+      'context': context
       }
     %}
     {# Pass vars through to include template. #}
-    {% include '@stanford_components/molecules/calendar-block/calendar-block.html.twig' with vars %}
+    {{ pattern('calendar_block', vars) }}
```
Comment (pookmish) [inline patterns/molecules/calendar-blocks/calendar-blocks.html.twig]: the `vars` is the same as `item`. could we just merge `context` into `item` during the `pattern()` call

File: patterns/molecules/callout-cards/callout-cards.html.twig [hunk 4]
```diff
@@ -4,17 +4,18 @@
...
       'link': item.link,
       'image': item.image,
-      'title': item.title
+      'title': item.title,
+      'context': context
       }
     %}
     {# Pass vars through to include template. #}
-    {% include '@stanford_components/molecules/callout-card/callout-card.html.twig' with vars %}
+    {{ pattern("callout_card", vars) }}
```
Comment (pookmish) [inline patterns/molecules/callout-cards/callout-cards.html.twig]: the `vars` is the same as `item`. could we just merge `context` into `item` during the `pattern()` call

File: patterns/molecules/collapsible-menu/collapsible-menu.html.twig [hunk 5]
```diff
@@ -7,22 +7,24 @@
...
       'link': item.link,
-      'name': item.name
+      'name': item.name,
+      'context': context
       }
     %}
 
     {# Pass vars through to include template. #}
-    {% include '@stanford_components/atoms/nav-item/nav-item.html.twig' with vars %}
+    {{ pattern("nav_item", vars) }}
```
Comment (pookmish) [inline patterns/molecules/collapsible-menu/collapsible-menu.html.twig]: the `vars` is the same as `item`. could we just merge `context` into `item` during the `pattern()` call

File: patterns/molecules/feature-blocks/feature-blocks.html.twig [hunk 6]
```diff
@@ -4,19 +4,20 @@
...
       'tag': item.tag,
       'description': item.description,
-      'more': item.more
+      'more': item.more,
+      'context': context
       }
     %}
     {# Pass vars through to include template. #}
-    {% include '@stanford_components/molecules/simple-block/simple-block.html.twig' with vars %}
+    {{ pattern('simple_block', vars) }}
```
Comment (pookmish) [inline patterns/molecules/feature-blocks/feature-blocks.html.twig]: the `vars` is the same as `item`. could we just merge `context` into `item` during the `pattern()` call

File: patterns/molecules/link-banner/link-banner.html.twig [hunk 7]
```diff
@@ -23,11 +22,12 @@
...
         'link': item.link,
         'title': item.title,
-        'subtitle': item.subtitle
+        'subtitle': item.subtitle,
+        'context': context
         } %}
 
         {# Pass vars through to include template. #}
-        {% include '@stanford_components/atoms/link-item/link-item.html.twig' with vars %}
+        {{ pattern('link_item',  vars) }}
```
Comment (pookmish) [inline patterns/molecules/link-banner/link-banner.html.twig]: the `vars` is the same as `item`. could we just merge `context` into `item` during the `pattern()` call

File: patterns/molecules/masonry-block/masonry-block.html.twig [hunk 8]
```diff
@@ -24,11 +23,12 @@
...
       {% for tag in tags %}
         {% set tag_vars = {
-          'tag': tag.tag
+          'tag': tag.tag,
+          'context': context
           }
         %}
       {# Pass vars through to include template. #}
-        {% include '@stanford_components/atoms/tag-item/tag-item.html.twig' with tag_vars %}
+        {{ pattern('tag_item', tag_vars) }}
```
Comment (pookmish) [inline patterns/molecules/masonry-block/masonry-block.html.twig]: the `tag_vars` is the same as `tag`. could we just merge `context` into `tag` during the `pattern()` call

File: patterns/molecules/masonry-blocks/masonry-blocks.html.twig [hunk 9]
```diff
@@ -19,17 +18,18 @@
...
       'link': card.link,
       'tags': card.tags,
-      'title': card.title
+      'title': card.title,
+      'context': context
       }
     %}
     {# Pass vars through to include template. #}
-    {% include '@stanford_components/molecules/masonry-block/masonry-block.html.twig' with card_vars %}
+    {{ pattern('masonry_block', card_vars) }}
```
Comment (pookmish) [inline patterns/molecules/masonry-blocks/masonry-blocks.html.twig]: the `card_vars` is almost the same as `card`. could we just merge the extras into `card` during the `pattern()` call

File: patterns/molecules/photo-tile/photo-tile.html.twig [hunk 10]
```diff
@@ -4,9 +4,10 @@
  * Photo tile pattern.
  */
 #}
-<div class="photo-tile tile-item">
+{% set attributes = attributes.addClass("photo-tile tile-item") %}
+<div{{ attributes }}>
   {% if link is not empty %}
```
Comment (pookmish) [inline patterns/molecules/photo-tile/photo-tile.html.twig]: No need for `if` now with `|openlink`

File: patterns/molecules/photo-tile/photo-tile.html.twig [hunk 11]
```diff
@@ -17,6 +18,6 @@
       </div>
     </div>
   {% if link is not empty %}
```
Comment (pookmish) [inline patterns/molecules/photo-tile/photo-tile.html.twig]: No need for this either

File: patterns/molecules/photo-tiles/photo-tiles.html.twig [hunk 12]
```diff
@@ -4,12 +4,18 @@
...
+{% if is_right %}
+  {% set attributes = attributes.addClass("is-right") %}
+{% endif %}
+{% if classes is not empty %}
+  {% set attributes = attributes.addClass(classes) %}
+{% endif %}
+
+<div{{ attributes }}>
   <div class="photo-tile tile-item tile-sizer">
     {% if link is not empty %}
```
Comment (pookmish) [inline patterns/molecules/photo-tiles/photo-tiles.html.twig]: no need for if

File: patterns/molecules/photo-tiles/photo-tiles.html.twig [hunk 13]
```diff
@@ -20,19 +26,20 @@
         </div>
       </div>
     {% if link is not empty %}
```
Comment (pookmish) [inline patterns/molecules/photo-tiles/photo-tiles.html.twig]: no need for if

File: patterns/molecules/photo-tiles/photo-tiles.html.twig [hunk 14]
```diff
@@ -20,19 +26,20 @@
...
     {% set vars = {
       'link': item.link,
-      'image': item.image
+      'image': item.image,
+      'context': context
       }
     %}
     {# Pass vars through to include template. #}
-    {% include '@stanford_components/atoms/photo-thumb/photo-thumb.html.twig' with vars %}
+    {{ pattern('photo_thumb', vars) }}
```
Comment (pookmish) [inline patterns/molecules/photo-tiles/photo-tiles.html.twig]: the `vars` is the same as `item`. could we just merge `context` into `item` during the `pattern()` call

File: patterns/molecules/simple-slats/simple-slats.html.twig [hunk 15]
```diff
@@ -5,7 +5,8 @@
  */
 #}
 {% set variants = variants|merge({'is_slat': true}) %}
```
Comment (pookmish) [inline patterns/molecules/simple-slats/simple-slats.html.twig]: Intersting. is this to pass context later?

File: patterns/organisms/section-calendar-blocks/section-calendar-blocks.html.twig [hunk 16]
```diff
@@ -27,11 +24,12 @@
...
           'day': item.day,
           'month': item.month,
-          'subtitle': item.subtitle
+          'subtitle': item.subtitle,
+          'context': context
           }
         %}
         {# Pass vars through to include template. #}
-        {% include '@stanford_components/molecules/calendar-block/calendar-block.html.twig' with vars %}
+        {{ pattern('calendar_block', vars) }}
```
Comment (pookmish) [inline patterns/organisms/section-calendar-blocks/section-calendar-blocks.html.twig]: the `vars` is the same as `item`. could we just merge `context` into `item` during the `pattern()` call

File: patterns/organisms/section-masonry-blocks/section-masonry-blocks.html.twig [hunk 17]
```diff
@@ -4,19 +4,17 @@
...
-<div class="section-masonry-blocks">
+{% set attributes = attributes.addClass("section-masonry-blocks") %}
+<div{{ attributes }}>
   {%  set nav_vars = {
-    'items': items
+    'items': items,
+    'context': context
   } %}
-  {% include '@stanford_components/molecules/collapsible-menu/collapsible-menu.html.twig' with nav_vars %}
+  {{ pattern('collapsible_menu', nav_vars) }}
```
Comment (pookmish) [inline patterns/organisms/section-masonry-blocks/section-masonry-blocks.html.twig]: could set this to `pattern('collapisble_menu', {'items': items, 'context': context})` but small reduction. not really needed

File: patterns/organisms/section-masonry-blocks/section-masonry-blocks.html.twig [hunk 18]
```diff
@@ -4,19 +4,17 @@
...
 
   {%  set card_vars = {
-    'cards': cards
+    'cards': cards,
+    'context': context
   } %}
-  {% include '@stanford_components/molecules/masonry-blocks/masonry-blocks.html.twig' with card_vars %}
+  {{ pattern('masonry_blocks', card_vars) }}
```
Comment (pookmish) [inline patterns/organisms/section-masonry-blocks/section-masonry-blocks.html.twig]: Same as line 13 comment

File: patterns/organisms/section-photo-mosaic-quotes/section-photo-mosaic-quotes.html.twig [hunk 19]
```diff
@@ -34,11 +28,12 @@
...
           'cite_name': tile.cite_name,
           'cite_title': tile.cite_title,
-          'quote': tile.quote
+          'quote': tile.quote,
+          'context': context
           }
         %}
         {# Pass vars through to include template. #}
-        {% include '@stanford_components/molecules/photo-tiles/photo-tiles.html.twig' with tile_vars %}
+        {{ pattern('photo_tiles', tile_vars) }}
```
Comment (pookmish) [inline patterns/organisms/section-photo-mosaic-quotes/section-photo-mosaic-quotes.html.twig]: the `tile_vars` is the same as `tile`. could we just merge `context` into `tile` during the `pattern()` call

File: patterns/organisms/section-photo-mosaic/section-photo-mosaic.html.twig [hunk 20]
```diff
@@ -30,11 +26,12 @@
...
           'items': tile.items,
           'is_right': tile.is_right,
-          'classes': tile.classes
+          'classes': tile.classes,
+          'context': context
           }
         %}
         {# Pass vars through to include template. #}
-        {% include '@stanford_components/molecules/photo-tiles/photo-tiles.html.twig' with tile_vars %}
+        {{ pattern('photo_tiles', tile_vars) }}
```
Comment (pookmish) [inline patterns/organisms/section-photo-mosaic/section-photo-mosaic.html.twig]: I think you know my thoughts at this point

File: patterns/molecules/simple-slats/simple-slats.html.twig (hunk 15 above)
Comment (sherakama) [inline patterns/molecules/simple-slats/simple-slats.html.twig]: This is forcing the variant through the context pass, yes.

**Comment by sherakama** [comment]
@pookmish,

Ok, all clean now.

**Pull request merged**